File: 240930/특정 수와 근접한 합/sum-close-to-particular-number.cpp
#include <iostream>
using namespace std;
int main() {
    int n,s,ans=10000000;
    cin>>n>>s;
    int arr[n];
    for(int i=0;i<n;i++)
        cin>>arr[i];
    for(int i=0;i<n;i++)
    {
        for(int j=0;j<n;j++)
        {
            if(i == j) continue;
            int sum=0;
            for(int k=0;k<n;k++)
            {
                if(k == i || k== j) continue;
                sum+=arr[k];
            }
            if(sum>s && sum-s < ans) ans = sum-s;
            else if(sum<s && s-sum < ans) ans = s-sum;
        }
    }
    cout<<ans;
    return 0;
}

File: 241018/특정 수와 근접한 합/sum-close-to-particular-number.py
n,s = map(int, input().split())

arr = list(map(int, input().split()))

answer = 0
for i in range(n):
    for j in range(i+1,n):
        minus_arr = arr[:i] + arr[i+1:j] + arr[j+1:]
        #print(i,j,minus_arr)
        
        # for k in range(len(minus_arr)):
        #     plus += minus_arr[k]
        plus = sum(minus_arr)
        #print(answer, plus)

        if abs(s - answer) >= abs(s - plus):
            answer = plus

print(abs(answer - s))